File: Assets/EZUnityTools/DestroyByTime.cs
/*
 * Author:      熊哲
 * CreateTime:  11/14/2017 7:09:31 PM
 * Description:
 * 
*/
using UnityEngine;

namespace EZComponent
{
    public class DestroyByTime : MonoBehaviour
    {
        [SerializeField]
        private float m_LifeTime = 1;
        public float lifeTime { get { return m_LifeTime; } set { m_LifeTime = value; } }

        void Start()
        {
            Destroy(gameObject, lifeTime);
        }
    }
}

File: Assets/Example/Script_Lua/SpaceShooter/DestroyByTime.lua
--[==[
- Author:       熊哲
- CreateTime:   11/15/2017 7:06:50 PM
- Orgnization:  #ORGNIZATION#
- Description:  
--]==]
local M = require("ezlua.module"):module()
----- CODE -----
function M.LCBinder(injector)
    local self = M:new()
    injector:Inject(self)
    self.gameObject = injector.gameObject
    CS.UnityEngine.Object.Destroy(self.gameObject, self.n_Lifetime)
    return self
end
----- CODE -----
return M
